### PR TITLE
Refactors/Removes Firebase from Package

### PR DESCRIPTION
**Pull request by peterfox**
Satisfies enhancement issue #1452 in removing all the code relying on Firebase's JWT package to instead rely on Lcobucci's package.

### CODE REVIEW

File: src/Guards/TokenGuard.php [hunk 1]
```diff
@@ -131,7 +131,7 @@ public function client(Request $request)
             );
         } elseif ($request->cookie(Passport::cookie())) {
             if ($token = $this->getTokenViaCookie($request)) {
-                return $this->clients->findActive($token['aud']);
+                return $this->clients->findActive($token['aud'][0]);
```
Comment (taylorotwell) [inline src/Guards/TokenGuard.php:134]: Why was this change needed?

Comment (peterfox) [reply src/Guards/TokenGuard.php:134]: The Lcobucci library breaks up audiences so you can have multiple values instead of a single string. When it parses the token it always produces an array. This is the default behaviour anyways. I don't know if there is potentially a way to stop that. It could in the Encoder class the audience should just be imploded into a string to replicate the way Firebase presents that value. I can do a bit more research into that.

I'll admit it as it stands it for people extending the method of the TokenGuard, then this change could be problematic for just a minor release.

**Comment by driesvints** [comment]
Hey @peterfox, appreciate the PR. I didn't realize we needed this much code to remove Firebase. Seems a bit problematic to maintain all of that code ourselves?

Also: I indeed think it's best that we do this in the next major release. Just to prevent breakges.

**Comment by peterfox** [comment]
@driesvints fair comments, it was more involved than I initially expected as well. I guess I could consider wrapping it up in a simple package as a drop in replacement for Firebase to reduce the code being added to passport itself.

Would you like me to look at opening up a PR for this against the master branch or would you rather we leave it for now?

**Comment by driesvints** [comment]
@peterfox. Thanks, I appreciate the offer but wouldn't it then be better if we kept Firebase? Swapping one package out for the other doesn't seems like the right choice. What was your original reason to remove Firebase? Also: is the other way around (swapping Lcobucci's JWT library out for Firebase) a better option?

**Comment by peterfox** [comment]
@driesvints the reason for removal was less redundancy. Laravel passport relies on [League's OAuth2 Server](https://github.com/thephpleague/oauth2-server/blob/master/composer.json) which in turn relies on Lcobucci's JWT package. So either way Lcobucci will be installed with an installation of Laravel passport. It can't be swapped out for Firebase because Passport is built around that OAuth2 Server package so that just isn't an option.

In the realms of Passport itself, keeping the Firebase package is fine and doesn't affect anything. In the larger scale of things, the Firebase package is a bit outdated and isn't actively maintained as well as the Lcobucci one, but as you've seen from the PR it's also a more complicated package with more configuration required.

This PR can probably be dropped for now, but I can see a future where the Firebase package ends up getting abandoned for more people picking up Lcobucci as it supports more in terms of security and functionality.

**Comment by driesvints** [comment]
Thanks @peterfox for that explanation. I do see @bshaffer is still maintaining the Firebase package actively so maybe this isn't needed in the end. I think we can leave things be for now. This was a good experiment though so thanks for that.

**Comment by peterfox** [comment]
@driesvints no worries. Was worthwhile looking into for myself also. Feel free to drop me a line about this if the situation changes.

**Comment by bshaffer** [comment]
It's true!! I'm maintaining it! We use it for all Google's client libraries.
